src/tags/include_tag.rs

```diff
@@ -29,25 +30,28 @@ impl Renderable for Include {
 fn parse_partial(name: &str, options: &LiquidOptions) -> Result<Template> {
     let content = options.include_source.include(name)?;
 
-    let tokens = tokenize(&content)?;
-    parse(&tokens, options).map(Template::new)
+    parse(&content, options).map(Template::new)
 }
 
 pub fn include_tag(
     _tag_name: &str,
-    arguments: &[Token],
+    arguments: &mut Iterator<Item=TagToken>,
     options: &LiquidOptions,
 ) -> Result<Box<Renderable>> {
-    let mut args = arguments.iter();
-
-    let name = match args.next() {
-        Some(&Token::StringLiteral(ref name)) => name,
-        Some(&Token::Identifier(ref s)) => s,
-        arg => return Err(unexpected_token_error("string", arg)),
+    let name = arguments.next().unwrap_or_else(|| panic!("Errors not implemented. Token expected."));
+    // TODO: make `name` a &str instead
+    let name = if let Ok(name) = name.expect_identifier() {
```

Comment:
Just to remind us of this todo

Reply:
Now that I look at this again, I realize that, at the end of the function, `name` would be casted into a `String` anyway. So there is no benefit in trying to keep it as a reference here.